iree/compiler/Codegen/LLVMGPU/Verifiers.cpp

```diff
@@ -36,11 +36,20 @@ LogicalResult verifyGPUMatmulSimtPassPipeline(
 
   if (linalg::BatchMatmulOp batchMatmulOp =
           dyn_cast<linalg::BatchMatmulOp>(op)) {
-    // First tile dimensions should be 1 for batched, use remaining dimensions
-    // for comparisons.
-    if (firstLevelTileSizes[0] != 1) {
-      op->emitError("Received first tile dimension of ")
-          << firstLevelTileSizes[0] << " instead of 1 for " << pipelineName;
+    // Inspect first tile dimensions separately for batched. It should be 1 for
+    // parallelizable loops and 0 for non-parallelizable. Continue with other
+    // dimensions for remaining comparisons.
+    if (cast<IREE::Flow::PartitionableLoopsInterface>(op).getPartitionableLoops(
+            kNumMaxParallelDims)[0] == 0) {  // The first dimension is
+      if (firstLevelTileSizes[0] != 1) {
+        op->emitError("Received first tile dimension of ")
+            << firstLevelTileSizes[0] << " instead of 1 for " << pipelineName;
+      }
+    } else {
+      if (firstLevelTileSizes[0] != 0) {
+        op->emitError("Received first tile dimension of ")
+            << firstLevelTileSizes[0] << " instead of 0 for " << pipelineName;
```

Comment:
I think we can allow both 0 and 1 as tile size if the highest dimension is 1, both would work so we may as well be more flexible

Reply:
Just FYI I hit an error with this piece of code in this PR https://github.com/google/iree/pull/8446 . I disabled the test that is modified below cause it wasnt clear what the expectation was (line 50 here - 42 before the change) should return an error, but is just silently proceeding. This probably need to be looked at again. Talked to @ThomasRaoux and decided to disable the test, and land PR #8446 . This needs to be fixed after the fact.  I can help with that as well.

Reply:
I'll merge after PR #8446. Are there additional changes to be made here beyond properly handling cases where the loop cannot be partitioned? ?

Reply:
The PR got merged, if you can rebase hopefully you should be able to re-enable the test and everything should be back :)